src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -311,6 +317,18 @@ private static bool InSymbolHeader(SyntaxNode matchingNode, int position)
                 return record.WithParameterList(record.ParameterList.WithParameters(updatedParameters).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
             }
 
+            if (updatedNode is ClassDeclarationSyntax { ParameterList: not null } @class)
+            {
+                var updatedParameters = UpdateDeclaration(@class.ParameterList.Parameters, signaturePermutation, CreateNewParameterSyntax);
+                return @class.WithParameterList(@class.ParameterList.WithParameters(updatedParameters).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
+            }
+
+            if (updatedNode is StructDeclarationSyntax { ParameterList: not null } @struct)
+            {
+                var updatedParameters = UpdateDeclaration(@struct.ParameterList.Parameters, signaturePermutation, CreateNewParameterSyntax);
+                return @struct.WithParameterList(@struct.ParameterList.WithParameters(updatedParameters).WithAdditionalAnnotations(changeSignatureFormattingAnnotation));
+            }
```

Comment:
could the above three 'if' statements be merged into one?  

Syntax question.  is 'ParameterList' added to TypeDeclSyntax, or only these three subclasses.  If the latter, could we consider actually moving up to TypeDeclSyntax and having it always be null for interfaces.  Or have interfaces allow parsing them out (for error recovery) and then report errors that it's not allowed there.  feels like it would help prevent a bunch of duplication in consumption code.

Reply:
> could the above three 'if' statements be merged into one?  

At the moment we do not have ```WithParameterList``` API on a base class and I prefer to not add it because it wouldn't be appropriate for all types derived from that base.

Reply:
> At the moment we do not have WithParameterList API on a base class and I prefer to not add it because it wouldn't be appropriate for all types derived from that base.

I'm totally fine with that for the current state of affairs.  Can this be noted though for when we bring this to API review?  I'd like to make an alternative case.  But that's def not blocking this work now.  Thanks! :)